File: src/Factoria/FactoriaCarretera.java
package src.Factoria;

import src.Bicicleta.Bicicleta;
import src.Bicicleta.BicicletaMontana;
import src.Carrera.Carrera;
import src.Carrera.CarreraMontana;

import java.util.ArrayList;

public class FactoriaCarretera implements FactoriaCarreraBicicleta{
    @Override
    public Carrera crearCarrera(ArrayList<Bicicleta> bicicletas) {
        return new CarreraMontana();
    }

    @Override
    public Bicicleta crearBicicleta(int id_bicicleta) {
        return new BicicletaMontana(id_bicicleta);
    }
}

File: src/Factoria/FactoriaCarretera.py
import random
from Factoria.FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta
from Bicicleta.BicicletaCarretera import BicicletaCarretera
from Carrera.CarreraCarretera import CarreraCarretera


class FactoriaCarretera(FactoriaCarreraYBicicleta):
    def crearCarrera(self, participantes):
        #Definir aqui la carrera
        #print('Creando carrera de Carretera')
        carrera = CarreraCarretera()

        for i in range(participantes):
            carrera.bicicletas.append(self.crearBicicleta())

        return carrera


    def crearBicicleta(self):
        #Definir aqui la bicicleta
        #print("Creando bicicleta de Carretera")
        id = str(random.randint(0,1000)).zfill(4)
        return BicicletaCarretera(id)